### PR TITLE
Fix volsize for cephfs and rbd

### PR DESCRIPTION
**Pull request by Madhu-1**
once the PVC is required and backend is created kubernetes sidecar will round off the values to a human-readable format, we have to create images with the same size to avoid volume size mismatch between PVC response and backend image size.

Signed-off-by: Madhu Rajanna <madhupr007@gmail.com>

### CODE REVIEW

File: pkg/cephfs/controllerserver.go [hunk 1]
```diff
@@ -84,17 +84,25 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
...
+	volOptions, err := newVolumeOptions(ctx, requestName, req.GetParameters(), secret)
 	if err != nil {
 		klog.Errorf(util.Log(ctx, "validation and extraction of volume options failed: %v"), err)
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
+	if req.GetCapacityRange() != nil {
+		volOptions.Size = util.RoundOffBytes(req.GetCapacityRange().GetRequiredBytes())
+	}
+	// TODO need to add check for 0 volume size
```
Comment (humblec) [inline pkg/cephfs/controllerserver.go]: I doubt we get a response at `0` size. It should have been blocked by CO.  have to double confirm though.

File: pkg/cephfs/controllerserver.go [hunk 2]
```diff
@@ -84,17 +84,25 @@ func (cs *ControllerServer) CreateVolume(ctx context.Context, req *csi.CreateVol
...
+
 	vID, err := checkVolExists(ctx, volOptions, secret)
 	if err != nil {
 		return nil, status.Error(codes.Internal, err.Error())
 	}
+
+	// TODO comapare request volume size and found volume sizem if requested is
```
Comment (humblec) [inline pkg/cephfs/controllerserver.go]: typo on sizem? or better to reword this todo.

**Comment by humblec** [comment]
@Madhu-1 please check the review comments.

File: pkg/cephfs/controllerserver.go (hunk 1 above)
Comment (Madhu-1) [reply pkg/cephfs/controllerserver.go]: not sure other CO will not process the vol size of `0`, but as per CSI spec it says volume size must not be negative but it can be zero. even earlier in cephfs, we are able to  create backend images with `0` size

File: pkg/cephfs/controllerserver.go (hunk 2 above)
Comment (Madhu-1) [reply pkg/cephfs/controllerserver.go]: updated

**Comment by humblec** [comment]
@ShyamsundarR @poornimag PTAL.

**Comment by humblec** [comment]
@ShyamsundarR @poornimag can you review this in priority?

File: pkg/util/util.go [hunk 3]
```diff
@@ -36,12 +37,43 @@ import (
...
+	roundedUp := volumeSizeBytes / allocationUnitBytes
+	if volumeSizeBytes%allocationUnitBytes > 0 {
+		roundedUp++
+	}
+	return roundedUp
+}
+
+// RoundOffBytes converts roundoff the size
+// 1.1Mib will be round off to 2Mib same for GiB
+// size less than will be round off to 1MiB
```
Comment (poornimag) [inline pkg/util/util.go]: nit: size less than 1MiB will be round off to 1MiB.

File: pkg/util/util.go [hunk 4]
```diff
@@ -36,12 +37,43 @@ import (
 const (
 	// MiB - MebiByte size
 	MiB = 1024 * 1024
+	GiB = MiB * 1024
 )
 
-// RoundUpToMiB rounds up given quantity upto chunks of MiB
-func RoundUpToMiB(size int64) int64 {
+// RoundOffVolSize rounds up given quantity upto chunks of MiB/GiB
+func RoundOffVolSize(size int64) int64 {
```
Comment (poornimag) [inline pkg/util/util.go]: RoundOffVolSize could call RoundOffBytes and convert it into number of MiBs and return?

Comment (Madhu-1) [reply pkg/util/util.go]: yes  this can be done, updated now

File: pkg/util/util.go (hunk 3 above)
Comment (Madhu-1) [reply pkg/util/util.go]: done
